File: foliage-rpc/src/main/java/org/lavenderx/foliage/nettyrpc/protocol/RpcRequest.java
package org.lavenderx.foliage.nettyrpc.protocol;

import lombok.Data;

import java.io.Serializable;

@Data
public class RpcRequest implements Serializable {
    private static final long serialVersionUID = -2073278080248168261L;

    private String requestId;
    private String className;
    private String methodName;
    private Class<?>[] parameterTypes;
    private Object[] parameters;
}

File: foliage-rpc/src/main/kotlin/io/foliage/netty/rpc/protocol/RpcRequest.kt
package io.foliage.netty.rpc.protocol

import java.io.Serializable

data class RpcRequest(val requestId: String) : Serializable {

    var className: String? = null
    var methodName: String? = null
    var parameterTypes: Array<Class<*>>? = null
    var parameters: Array<Any>? = null

    companion object {
        private val serialVersionUID = -2073278080248168261L
    }
}